### PR TITLE
Gutenberg: Use Fullscreen mode in Calypsoify 

### LINKED ISSUE #27937: Gutenberg: Consider using the new close icon with tooltip
Looks like Core Gutenberg got a new close icon and tooltip for _Full Screen_ mode in the recent update. Here's what it looks like when you hover over it:

image

I think we may want to try swapping out our custom Close button, which looks like this today:

image

Since our button doesn't always take you back to posts, we should be sure to use the Tooltip to be more specific about what the button will do; i.e. Back to stats, or Back to your site, or Back to post, etc

### PR DESCRIPTION
**Pull request by mmtr**
#### Changes proposed in this Pull Request:

Calypsoify uses now the Gutenberg Fullscreen Mode instead of injecting the close button and including custom CSS.

We realized that the Gutenberg Fullscreen Mode is providing a UI very similar to Calypso, so by using it we can simplify the code and make it more maintainable. 

~This PR also includes a small fix for an 500 error raised when Jetpack is checking if Gutenberg is enabled.~ Fixed on #10389.

Fixes Automattic/wp-calypso#27937

#### Testing instructions:

* Edit a post in Gutenberg
* Append `&calypsoify=1` to the URL and reload the post editor.
* Make sure that the editor is displayed in Fullscreen Mode.
* Check that there are no options for disabling the Fullscreen Mode when clicking on the "More" button.
* Confirm that the back button uses now an icon and it redirects to the posts list in Calypso when clicking on it.

#### Screenshots:

Before
screen shot 2018-10-22 at 14 12 45

After
screen shot 2018-10-22 at 15 17 53

#### Proposed changelog entry for your changes:

I don't think this is needed given Calypsoify is mainly used internally. Anyway, if needed, this is my proposed changelog:

> Calypsoify uses now the Gutenberg Fullscreen Mode instead of injecting the close button and including custom CSS.

### CODE REVIEW

File: modules/calypsoify/style-gutenberg.css
Comment (Copons) [inline modules/calypsoify/style-gutenberg.css]: This PR is more oriented at making `Calypsoified Gutenberg` and `Gutenberg in Calypso` as identical as possible, but in this particular case, I'd say let's keep the group visible even if it's not there in Calypso, and only hide the "Fullscreen Mode" item.

Also, notice that _now_ it's the last child, but that could change.
Off the top of my head there is no simple selector there, but I might be very wrong. 🙂

**Comment by Copons** [comment]
Let me add a bit of history behind this PR.

When I wrote the code for Calypsoify Gutenberg, there was no fullscreen mode yet (or maybe there was but I didn't notice it), and the whole idea was to make Gutenberg look as much as possible as Calypso.
I've spent time styiling the main UI components (buttons, notices, etc.) to make them look as native Calypso components.

Meanwhile, Calypso-side, it had been decided that Gutenberg would have kept its style for now.
This caused the weirdness that Calypsoified Gutenberg looked more like Calypso than Gutenberg in Calypso. (I know, not simple to follow 😅)

When we learned that Gutenberg had a new fullscreen mode with a nice "Back" icon, we went to replace the Calypsoified Gutenberg's "Close" button with that icon.
Only then we realized that the fullscreen mode looked very much like Gutenberg in Calypso itself. Like, a lot.

All the overrides needed to hide the masterbar, the sidebar, etc. were there, a mere Redux dispatch away.

So this PR simply dispatches that action, and removes (most) of the overrides as they are not needed anymore.

File: modules/calypsoify/style-gutenberg.css
Comment (mmtr) [reply modules/calypsoify/style-gutenberg.css]: Thanks for that! I've just committed a change handling this 🙂

I thought this view mode selector was like a radio button (you can only select one of them), but I realize that you can activate all of them at the same time, so it makes sense to only hide the "Fullscreen Mode" option.

Regarding the CSS selector using `:last-child`. I don't like it either, but the HTML rendered by Gutenberg doesn't provide any other class we can use for identifying this element:

screen shot 2018-10-22 at 19 10 43

**Review by Copons — DISMISSED** [review]
This LGTM!
We should be well in time for the 6.7 release! ✨

**Comment by jeherve** [comment]
c'ing @georgestephanis for a review on this since he's currently working on other Calypsoify improvements in #10371

**Comment by Copons** [comment]
Oh very cool!
I think that if we get to merge this, and rebase #10371, it will become considerably smaller.
On the other hand, it could be easier to just close this and reopen it after #10371 merges.
Either way sounds good to me, @georgestephanis

**Comment by gwwar** [comment]
@jeherve Can we still target this for the 6.7 release? @Copons any blockers here? I can give this a spin as well.

**Review by gwwar — DISMISSED** [review]
Feels pretty solid! Unless folks find blockers this should be good to go. Excellent line count @mmtr! 💖

screen shot 2018-10-26 at 10 51 06 am

**Review by brbrr — COMMENTED** [review]
I noticed one small issue. Header toolbar is not fits into the screen in Safari due to `Tips` pop-up. After dismissing that pop-up - page loads without issues. Is it something that could be fixed in this PR?

Besides that - all looks good 👍 


jetpack6test mystagingwebsite com at pressable wordpress 2018-10-26 21-40-34

**Comment by mmtr** [comment]
@brbrr are you referring to the tip misplacement because it is not placed under the '+' icon? It looks like that problem is present in Gutenberg itself, so I opened an issue there: WordPress/gutenberg#11194

**Review by jeherve — DISMISSED** [review]
This works well for me! 🚢

**Comment by brbrr** [comment]
@mmtr Sorry, that's not clearly visible from the screenshot. I referring to misplaced _toolbar buttons_: `+`, `back`, `forward`, etc.

**Review by brbrr — DISMISSED** [review]
:shipit: :)

**Comment by Copons** [comment]
Sorry folks, this accidentally breaks the Revisions view. I'll push a hotfix in a bit, but meanwhile do not merge please! 🙇

**Comment by Copons** [comment]
Again, sorry about this.
I was too hasty in deleting all that CSS, and forgot that part of it was applied to the Revisions page, which doesn't load Gutenberg, and so of course doesn't have `wp.data.select`.

7360cde restore the CSS overrides of the Revisions page, and guard against undefined `wp.data.select` which killed it.

Testing instructions:

- Load a Gutenberg post, with `?calypsoify=1`
- In order to get some revisions of the post: do some changes, save; do some other changes; update.
- Click on the new Revisions link in the Gutenberg's Document sidebar.
- Make sure the Revisions page is calypsoified as well (doesn't have any wp-admin sidebar and masterbar), and that clicking on the "back to editor" links redirects to a still calypsoified version of Gutenberg (no need to have the `?calypsoify=1` anymore; one way to check if Calypsoify is active is to check the Close URL, it should point to Calypso).

**Review by jeherve — DISMISSED** [review]
Nice catch! 👍

**Comment by Copons** [comment]
I feel so bad, but I've had to sneak in one last change. 😭 

ae88b29 adds back the CSS needed to hide the "Manage Reusable Blocks" links from the editor.
More context in 9-gh-calypsoify

Testing instructions:

- Create a reusable block (block toolbar -> dots menu -> "Add to reusable blocks").
- Open the Blocks inserter, and expand the "Reusable" section (should be the last one). 
- Make sure the "Manage All Reusable Blocks" link is hidden.

**Review by kraftbj — APPROVED** [review]

**Review by jeherve — APPROVED** [review]
This seems to work for me. Merging.

**Pull request merged**
